components/org.wso2.carbon.identity.api.user.totp/org.wso2.carbon.identity.api.user.totp.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/totp/v1/core/TOTPService.java

```diff
@@ -207,11 +240,12 @@ public TOTPResponseDTO validateTOTP(int verificationCode) {
                             .setKeyRepresentation(encoding);
             TOTPAuthenticatorCredentials totpAuthenticator =
                     new TOTPAuthenticatorCredentials(configBuilder.build());
-            String secretKey = getSecretKey().getSecret();
+            String secretKey = getVerifySecretKey();
             if (log.isDebugEnabled()) {
                 log.debug("Validating TOTP verification code for the user: " + username);
             }
-            totpResponseDTO.setIsValid(totpAuthenticator.authorize(secretKey, verificationCode));
```

Comment:
I prefer if we can write this method in a way where TOTP authenticator resolve the secret instead of we are passing the secret back to authenticator by API level.

Reply:
can we check if that is possible? If so, we can completely remove `getVerifySecretKey` and call TOTP authenticator method to verify and authenticate

Reply:
In that case, `secretKey` and `verifySecretKey` both will be resolved by TOTP authenticator. If `verifySecretKey` is validated, we can move `verifySecretKey` to `secretKey` claim. Also, I think, we should introduce a new method for this as validateTOTP method should not update the secretKey